common/libs/wallet/deterministic/ledger.ts

```diff
@@ -1,80 +1,142 @@
-import ledger from 'ledgerco';
+import TransportU2F from '@ledgerhq/hw-transport-u2f';
+import LedgerEth from '@ledgerhq/hw-app-eth';
 import EthTx, { TxObj } from 'ethereumjs-tx';
 import { addHexPrefix, toBuffer } from 'ethereumjs-util';
-import { DeterministicWallet } from './deterministic';
+import { HardwareWallet, ChainCodeResponse } from './hardware';
 import { getTransactionFields } from 'libs/transaction';
 import { IFullWallet } from '../IWallet';
 import { translateRaw } from 'translations';
+import EnclaveAPI, { WalletTypes } from 'shared/enclave/client';
 
-export class LedgerWallet extends DeterministicWallet implements IFullWallet {
-  private ethApp: ledger.eth;
+export class LedgerWallet extends HardwareWallet implements IFullWallet {
+  public static async getChainCode(dpath: string): Promise<ChainCodeResponse> {
+    if (process.env.BUILD_ELECTRON) {
+      return EnclaveAPI.getChainCode({
+        walletType: WalletTypes.LEDGER,
+        dpath
+      });
+    }
+
+    return makeApp()
+      .then(app => app.getAddress(dpath, false, true))
+      .then(res => {
+        return {
+          publicKey: res.publicKey,
+          chainCode: res.chainCode as string
```

Comment:
What's the reason for the typecast here?

Reply:
`app.getAddress` thinks this can be undefined, but that's only if you pass `false` as third arg or don't pass a third arg. This could be fixed in our typings by looking at that arg in the typing, I think we discussed this over DM a bit. I can't remember where we left that.